src/data/list/basic.lean

```diff
@@ -4658,4 +4658,24 @@ begin
       apply lt_of_succ_lt_succ hi, } },
 end
 
+/-- Backwards recursor on lists, inducts by appending elements to the end of the list -/
+def backwards_rec {α : Type u} {P : list α → Sort v} (hPempty : P [])
```

Comment:
I never remember what `elab_as_eliminator` does, but should that be added to this or the other def?

Reply:
From the reference manual:
> The last attribute, `[elab_as_eliminator]`, is commonly used for eliminators like recursors, induction principles, and `eq.subst`. It uses a separate heuristic to infer higher-order parameters. 
Should it be both?

Reply:
For reference, `list.rec_on` does not seem to be annotated in this way. `list.rec` also isn't, but that seems to be annotated oddly by `#print` with an imaginary `eliminator` keyword, so I don't know how to use that for reference.

Reply:
`rec_on` and `rec` are special since they are auto-generated.  `backwards_rec` should be `@[elab_as_eliminator]`.

Reply:
This is causing problems with defining `backwards_rec_nil`, `@[elab_as_eliminator]` is causing it to complain at `backwards_rec hPnil hPind []`, saying I cannot use metavariables

Reply:
This might be because it doesn't know the type of the list's elements: perhaps using `([] : list \a)` instead of `[]` could work?